airflow/providers/samba/hooks/samba.py

```diff
@@ -59,8 +59,8 @@ def __init__(self, samba_conn_id: str = default_conn_name, share: str | None = N
 
         connection_cache: dict[str, smbprotocol.connection.Connection] = {}
 
-        self._host = conn.host
-        self._share = share or conn.schema
+        self._host = conn.host or ""
+        self._share = share or conn.schema or ""
```

Comment:
Can you please add a unit test for this?

Reply:
Done!